src/reflect/scala/reflect/internal/transform/Erasure.scala

```diff
@@ -142,8 +146,8 @@ trait Erasure {
           // this replaces each typeref that refers to an argument
           // by the type `p.tpe` of the actual argument p (p in params)
           else apply(mt.resultType(mt.paramTypes)))
-      case RefinedType(parents, decls) =>
-        apply(mergeParents(parents))
+      case rt : RefinedType =>
+        apply(mergeParents(rt.normalize.parents)) // RefinedType#normalize
```

Comment:
`JavaErasureMap` overrides `mergeParents`:

```scala
    /** In java, always take the first parent.
     *  An intersection such as `Object with Trait` erases to Object.
     */
    def mergeParents(parents: List[Type]): Type =
      if (parents.isEmpty) ObjectTpe
      else parents.head
```

Rather than:

```scala
    /** In scala, calculate a useful parent.
     *  An intersection such as `Object with Trait` erases to Trait.
     */
    def mergeParents(parents: List[Type]): Type =
      intersectionDominator(parents)
```

The erasure type map is run (just) prior to `phase.erasedTypes`, so `RefinedType.normalize` leans on an internal cache:

```scala
    final override def normalize: Type =
      if (phase.erasedTypes) normalizeImpl
      else {
        if (normalized eq null) normalized = normalizeImpl
        normalized
      }
```

If this cache were null at the time we called `normalize`. If `normalize` is not an identity (ie we need to flatten nested refinements), there is a small inefficiency in:
  - computing the flat list of normalized parents and creating a new `RefinedType` with them and the `decls`
  - ... only to extract the `parents` and discard the rest in `ErasureMap`
  - ... only to use the first parent in `JavaErasureMap`.

I believe most of the time `RefinedType.normalize` is an identity, so we're unlikely to see a performance win from this.